python_basics_writing_python_code/python_basics_writing_python_code.md

```diff
@@ -0,0 +1,921 @@
+
+    * `.pop()` removes a specific element by **index**).
+
+     @sage
+     <div class="python_link">
+     <lia-keep>
+     <script type="text/x-sage">
+
+     print(produce.pop(0))
+
+     </script>
+     </lia-keep>
+     </div>
+
+   * `.remove()` removes a specific element by **value**.
+
+    @sage
+    <div class="python_link">
+    <lia-keep>
+    <script type="text/x-sage">
+
+    produce.remove("apple")
+print(produce)
+
+    </script>
+    </lia-keep>
+    </div>
+
+  * `del` deletes objects, and can be used as another way to remove elements of a list by index. The syntax will be a little different, because it is a keyword, not a list method. It is useful because it allows you to delete list elements in a slice. Slicing uses subsetting notation. the first index listed is **inclusive**, or included in the slice; this is followed by a colon; the second index listed is **exclusive**, or excluded from the slice; finally, these numbers are surrounded by square brackets.
+
+    @sage
+    <div class="python_link">
+    <lia-keep>
+    <script type="text/x-sage">
+
+    del produce[1:3]
+print(produce)
+
+    </script>
+    </lia-keep>
+    </div>
+
+  * `.clear()` removes everything from the list.
+
+    @sage
+    <div class="python_link">
+    <lia-keep>
+    <script type="text/x-sage">
+
+    produce.clear()
+print(produce)
+
+    </script>
+    </lia-keep>
+    </div>
+
+  * `.append()` adds an element to the end of a list.
+
+    @sage
+    <div class="python_link">
+    <lia-keep>
+    <script type="text/x-sage">
+
+    produce.append("pear")
+print(produce)
+
+    </script>
+    </lia-keep>
+    </div>
+
+  * `.insert()` adds an element to a specific position.
+
+    @sage
+    <div class="python_link">
+    <lia-keep>
+    <script type="text/x-sage">
+
+    produce.insert(0, "kale")
+print(produce)
+
+    </script>
+    </lia-keep>
+    </div>
+
+  * `.extend()` adds another list to your list (or another iterable object, but don't worry about that for now).
+
+    @sage
+    <div class="python_link">
+    <lia-keep>
+    <script type="text/x-sage">
+    berries= ["strawberry", "blueberry", "raspberries"]
+produce.extend(berries)
+print(produce)
+
+    </script>
+    </lia-keep>
+    </div>
+
+
+You'll notice that with each step, our list `produce` has been permanently changed-- in order to "start over" with the original list, we simply need to re-run the first code chunk.
+
+<div class = "warning">
+The code cells above are linked, meaning that all of the cells "remember" the code that has been run previously (as they would be in a [notebook](https://liascript.github.io/course/?https://raw.githubusercontent.com/arcus/education_modules/update-intro-to-python/intro_to_python/intro_to_python.md#8)). Because of this, if you run them out of order, you can get unexpected results or even errors! If things aren't working like they should, re-run the first code cell on the page or hit your refresh button.
```

Comment:
I'm going to borrow this warning and put it lots more places! really good warning!

Reply:
Go for it! It trips people up in Jupyter too, so I imagine there are a lot of places we can use a warning like this :)